src/main/api/fileStore.ts

```diff
@@ -0,0 +1,79 @@
+import path from 'path'
+
+import { app, ipcRenderer, ipcMain } from 'electron'
+import * as fs from 'fs-extra'
+
+import { ApiFileStoreService, StoreFileData, StoreFileName } from '../../shared/api/types'
+import { getStoreFilesIPCMessages } from '../../shared/ipc/fileStore'
+
+const APP_NAME = app?.name ?? 'ASGARDEX'
+
+const APP_DATA_DIR = path.join(app?.getPath('appData') ?? './testdata', APP_NAME)
+export const STORAGE_DIR = path.join(APP_DATA_DIR, 'storage')
+
+// If `fullFilePathname' does not exist, `fs.remove` silently does nothing.
+// @see https://github.com/jprichardson/node-fs-extra/blob/master/docs/remove.md
+const removeFile = async (fullFilePathname: string) => fs.remove(fullFilePathname)
+
+const isFileExist = async (fullFilePathname: string) => fs.pathExists(fullFilePathname)
+
+const getFileContent = async <T>(fullFilePathname: string, defaultValue: T): Promise<T> => {
+  // If wile does not exist create a new one with defaultValue as content
+  if (!(await isFileExist(fullFilePathname))) {
+    await fs.writeJSON(fullFilePathname, { ...defaultValue })
+    return Promise.resolve(defaultValue)
+  }
+
+  try {
+    // Try to read and parse appropriate JSON store-file
+    const fileContent = await fs.readJSON(fullFilePathname)
+    // Combine file content with provided default value
+    return Promise.resolve({ ...defaultValue, ...fileContent })
+  } catch {
+    // Resolve with default value in case of any error
+    return Promise.resolve(defaultValue)
+  }
+}
+/**
+ * Replaces store-file's content with partial provided data
```

Comment:
[nitpick] ^ It does not replace, it appends new data to existing data in a file

Reply:
forgot to change it. i'd say `combines`. do you mind `combines` ?